Mathlib/LinearAlgebra/ExteriorPower/Basic.lean

```diff
@@ -0,0 +1,204 @@
+expresses the universal property of the exterior power as a
+linear equivalence `(M [⋀^Fin n]→ₗ[R] N) ≃ₗ[R] ⋀[R]^n M →ₗ[R] N` between
+alternating maps and linear maps from the exterior power.
+
+-/
+
+open scoped TensorProduct
+
+universe u v uM uN uN' uN'' uE uF
+
+variable (R : Type u) (n : ℕ) {M : Type uM} {N : Type uN} {N' : Type uN'} {N'' : Type uN''}
+  [CommRing R] [AddCommGroup M] [Module R M] [AddCommGroup N] [Module R N] [AddCommGroup N']
+  [Module R N'] [AddCommGroup N''] [Module R N'']
+
+variable {K : Type v} {E : Type uE} {F : Type uF} [Field K] [AddCommGroup E] [Module K E]
+  [AddCommGroup F] [Module K F]
+
+namespace exteriorPower
+
+open Function
+
+/-! The canonical alternating map from `Fin n → M` to `⋀[R]^n M`. -/
+
+/-- `exteriorAlgebra.ιMulti` is the alternating map from `Fin n → M` to `⋀[r]^n M`
+induced by `exteriorAlgebra.ιMulti`, i.e. sending a family of vectors `m : Fin n → M` to the
+product of its entries. -/
+def ιMulti : M [⋀^Fin n]→ₗ[R] (⋀[R]^n M) :=
+  (ExteriorAlgebra.ιMulti R n).codRestrict (⋀[R]^n M) fun _ =>
+    ExteriorAlgebra.ιMulti_range R n <| Set.mem_range_self _
+
+@[simp] lemma ιMulti_apply_coe (a : Fin n → M) : ιMulti R n a = ExteriorAlgebra.ιMulti R n a := rfl
+
+variable (M)
+/-- The image of `ExteriorAlgebra.ιMulti R n` spans the `n`th exterior power. Variant of
+`ExteriorAlgebra.ιMulti_span_fixedDegree`, useful in rewrites. -/
+lemma ιMulti_span_fixedDegree :
+    Submodule.span R (Set.range (ExteriorAlgebra.ιMulti R n)) = ⋀[R]^n M :=
+  ExteriorAlgebra.ιMulti_span_fixedDegree R n
+
+/-- The image of `exteriorPower.ιMulti` spans `⋀[R]^n M`. -/
+lemma ιMulti_span :
+    Submodule.span R (Set.range (ιMulti R n)) = (⊤ : Submodule R (⋀[R]^n M)) := by
+  apply LinearMap.map_injective (Submodule.ker_subtype (⋀[R]^n M))
+  rw [LinearMap.map_span, ← Set.image_univ, Set.image_image]
+  simp only [Submodule.coe_subtype, ιMulti_apply_coe, Set.image_univ, Submodule.map_top,
+    Submodule.range_subtype]
+  exact ExteriorAlgebra.ιMulti_span_fixedDegree R n
+
+namespace presentation
+
+/-- The index type for the relations in the standard presentation of `⋀[R]^n M`,
+in the particular case `ι` is `Fin n`. -/
+inductive Rels (ι : Type*) (M : Type*)
+  | add (m : ι → M) (i : ι) (x y : M)
+  | smul (m : ι → M) (i : ι) (r : R) (x : M)
+  | alt (m : ι → M) (i j : ι) (hm : m i = m j) (hij : i ≠ j)
+
+/-- The relations in the standard presentation of `⋀[R]^n M` with generators and relations. -/
+@[simps]
+noncomputable def relations (ι : Type*) [DecidableEq ι] (M : Type*)
+    [AddCommGroup M] [Module R M] :
+    Module.Relations R where
+  G := ι → M
+  R := Rels R ι M
+  relation r := match r with
+    | .add m i x y => Finsupp.single ((update m i x)) 1 +
+        Finsupp.single ((update m i y)) 1 -
+        Finsupp.single ((update m i (x + y))) 1
+    | .smul m i r x => Finsupp.single ((update m i (r • x))) 1 -
+        r • Finsupp.single ((update m i x)) 1
```

Comment:
style nit: all these parens are unnecessarily doubled.

Reply:
Thanks for this observation!